scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/ModelValidatorContext.java

```diff
@@ -72,6 +84,7 @@ public ModelValidatorContext(TaskFlowJob job) {
 
         spELVariables = new SpELVariables(jobVariablesValues);
         spelContext = new StandardEvaluationContext(spELVariables);
+        spelContext.setTypeLocator(new RestrictedTypeLocator());
```

Comment:
public ModelValidatorContext(StandardEvaluationContext context) does not require `spelContext.setTypeLocator(new RestrictedTypeLocator());` ?

I suggest you yo use cascading constructors instead of using the same code in the different constructors.

Reply:
After discussion with Amine is not relevant to put ` spelContext.setTypeLocator(new RestrictedTypeLocator());` in `ModelValidatorContext(StandardEvaluationContext context)` constructor

Reply:
However this can be replaced by
```java
 private ModelValidatorContext(Map<String, Serializable> variablesValues) {
    spELVariables = new SpELVariables(jobVariablesValues);
    spelContext = new StandardEvaluationContext(spELVariables);
    spelContext.setTypeLocator(new RestrictedTypeLocator());
}

 public ModelValidatorContext(Task task) {
     this(job.getVariables().values().stream().collect(Collectors.toMap(TaskVariable::getName,TaskVariable::getValue)));
}

public ModelValidatorContext(TaskFlowJob job) {
    this(job.getVariables().values().stream().collect(Collectors.toMap(JobVariable::getName,JobVariable::getValue)));
}
```

Reply:
@aminelouati I updated the upper code now it should work.

Reply:
Thanks @MaelAudren